lib/devise_saml_authenticatable/saml_mapped_attributes.rb

```diff
@@ -10,28 +10,43 @@ def saml_attribute_keys
     end
 
     def resource_keys
-      @attribute_map.values
+      @attribute_map.values.map { |h| h["resource_key"] }
     end
 
     def value_by_resource_key(key)
       str_key = String(key)
 
       # Find all of the SAML attributes that map to the resource key
-      attribute_map_for_key = @attribute_map.select { |_, resource_key| String(resource_key) == str_key }
+      attribute_map_for_key = @attribute_map.select { |_, config| String(config["resource_key"]) == str_key }
 
+      # When an attribute is "multi", return the entire array
+      # When the attribute is "single", return the first value
       saml_value = nil
-
-      # Find the first non-nil value
-      attribute_map_for_key.each_key do |saml_key|
-        saml_value = value_by_saml_attribute_key(saml_key)
-
-        break unless saml_value.nil?
+      attribute_map_for_key.each_pair do |saml_key, config|
+        if config["attribute_type"] == "multi"
+          saml_value ||= []
+          saml_value << value_by_saml_attribute_key(saml_key, config)
+        else
+          saml_value = value_by_saml_attribute_key(saml_key, config)
+          return saml_value unless saml_value.nil?
```

Comment:
I'm not actually sure if I need these changes, as value_by_saml_attribute_key should just return an array